contracts/Nexus.sol

```diff
@@ -335,14 +338,13 @@ contract Nexus is INexus, BaseAccount, ExecutionHelper, ModuleManager, UUPSUpgra
     /// and return 0xffffffff as a result.
     function checkERC7739Support(bytes32 hash, bytes calldata signature) public view virtual returns (bool) {
         unchecked {
-            if (signature.length == uint256(0)) {
-                // Forces the compiler to optimize for smaller bytecode size.
-                if (uint256(hash) == (~signature.length / 0xffff) * 0x7739) {
-                    SentinelListLib.SentinelList storage validators = _getAccountStorage().validators;
-                    address next = validators.entries[SENTINEL];
-                    while (next != ZERO_ADDRESS && next != SENTINEL) {
-                        if (IValidator(next).isValidSignatureWithSender(msg.sender, hash, signature) == SUPPORTS_ERC7739) return true;
-                    }
+            // Forces the compiler to optimize for smaller bytecode size.
+            if (uint256(hash) == (~signature.length / 0xffff) * 0x7739) {
```

Comment:
curious what does ~ do?

Reply:
it is bitwise negation. so it makes 0xfff..fff out of 0.